File: filtro2.py
import numpy as np
from scipy import signal

a = [1,2,3,4]
b = [1,2,3,4]

Na = len(a)
input_sample = np.linspace(1,10,10,endpoint=True)
Nx = len(input_sample)
yy = signal.lfilter(b,a,input_sample)

u = np.zeros(Na)

for n in range(0,Nx,1):
    x = input_sample[n]
    temp1 = 0
    temp2 = 0
    for k in range(1,Na,1):
        temp1 = temp1 - a[k]*u[k]
        temp2 = temp2 + b[k]*u[k]
    u[0] = x + temp1
    y = u[0]*b[0]+temp2
    
    for m in range(Na-1,0,-1):
        u[m] = u[m-1]
        
    print(y)
    print(yy)
    print("Oprime enter")

File: filtro2.m
clear; close all; clc;
a = [1,2,3,4];
Na = length(a);
b = [2,3,4,5];
input_sample = 1:10;
Nx = length(input_sample);
yy = filter(b,a,input_sample); %Filtrado IIR con procesamiento por bloque
u = zeros(1,Na);

for n=1:Nx
    x=input_sample(n);
    temp1=0;
    temp2=0;
    y=0;
    for k=2:Na
        temp1=temp1-a(k)*u(k);
        temp2=temp2+b(k)*u(k);
    end
    u(1) = x+temp1;
    y = u(1)*b(1)+temp2;
    for m=Na:-1:2
        u(m)=u(m-1);
    end
    yy
    y
    pause
end